source/core/Events.cpp

```diff
@@ -943,7 +943,7 @@ VIREO_FUNCTION_SIGNATUREV(WaitForEventsAndDispatch, WaitForEventsParamBlock)
                             eventRegRefnumPtr[regRefIndex-1].Type()->GetSubElement(0) : nullptr) : 0;
 
     for (Int32 inputTuple = 0; inputTuple < numTuples; ++inputTuple) {
-        UInt32 eventSpecIndex = *arguments[inputTuple].eventSpecIndex;
+        IntIndex eventSpecIndex = *arguments[inputTuple].eventSpecIndex;
         if (eventSpecIndex >= esCount) {
```

Comment:
And either change the type of esCount to be UInt32 or cast here.

Reply:
esCount is assigned from the value of a call to TypeCommon::SubElementCount, which returns an Int32. For the reason in my above comment I would prefer to keep this a signed int.

Reply:
This is an argument to WaitForEvents which comes from via code and is checked at run-time.
Changing it to Int32 would require also checking to see it's non-negative to validate it.
So this warning fix would slow run-time; that's not a good trade-off.

Instead, make esCount match.

`UInt32 esCount = UInt32(eventSpecType->SubElementCount());`